### PR TITLE
feature/SPRIND-83

### PR DESCRIPTION
**Pull request by zoemaas**

### CODE REVIEW

File: packages/anomaly-detection/src/types/IAnomalyDetection.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import {IAgentContext, IPluginMethodMap} from '@veramo/core'
+import {AnomalyDetectionStoreArgs, IAnomalyDetectionStore} from "@sphereon/ssi-sdk.anomaly-detection-store";
+import {Optional} from "nx/src/project-graph/plugins";
```
Comment (sanderPostma) [inline packages/anomaly-detection/src/types/IAnomalyDetection.ts:3]: /src import

**Review by sanderPostma — CHANGES_REQUESTED** [review]
Found bad import

**Review by sanderPostma — APPROVED** [review]
LGTM when import is fixed.
When this is not possible for some reason, add a comment

File: packages/anomaly-detection/src/types/IAnomalyDetection.ts (hunk 1 above)
Comment (zoemaas) [reply packages/anomaly-detection/src/types/IAnomalyDetection.ts:3]: It looks like a mixed it up with Java :see_no_evil:

Comment (zoemaas) [reply packages/anomaly-detection/src/types/IAnomalyDetection.ts:3]: Replaced it with a Partial

File: packages/anomaly-detection-store/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  <br>
+</h1>
+
+---
+
+**Warning: This package still is in very early development. Breaking changes without notice will happen at this point!**
+
+---
+
+A Veramo anomaly detection store plugin. This plugin stores the result of the location lookup which is part of the anomaly detection plugin
```
Comment (nklomp) [inline packages/anomaly-detection-store/README.md:15]: Please stop using the word Veramo everywhere. This really is not a Veramo plugin

File: packages/anomaly-detection-store/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+## Usage
+
+### Adding the plugin to an agent:
+
+```typescript
+import { AnomalyDetectionStore } from '@sphereon/ssi-sdk.anomaly-detection'
+
+const agent = createAgent<IAnomalyDetectionStore>({
+  plugins: [
+    new AnomalyDetectionStore({
```
Comment (nklomp) [inline packages/anomaly-detection-store/README.md:30]: This should be renamed to GeoLocationStore

File: packages/anomaly-detection-store/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      geoIpDBPath: './GeoLite2-Country.mmdb',
+    }),
+  ],
+})
+```
+
+### Lookup a location:
+
+```typescript
+const result = await agent.lookupLocation({
```
Comment (nklomp) [inline packages/anomaly-detection-store/README.md:40]: Agent plugin should prefix something, to ensure that a developer can actually make head or tails about the 100+ methods available in a plugin.

So geoLookupLocation makes sense

File: packages/anomaly-detection-store/src/agent/AnomalyDetectionStore.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  'anomalyDetectionStoreRemoveLocation',
+  'anomalyDetectionStoreClearAllLocations',
+  'anomalyDetectionStoreGetLocation',
+  'anomalyDetectionStoreDefaultLocationStore'
+]
+
+/**
+ * {@inheritDoc IAnomalyDetectionStore}
+ */
+export class AnomalyDetectionStore implements IAgentPlugin {
```
Comment (nklomp) [inline packages/anomaly-detection-store/src/agent/AnomalyDetectionStore.ts:28]: Please rename to GeoLocationStore. This is only about Geo locations, which happens to be part of anomaly detection. We will not store other data in this plugin in the future. 

So rename the class and also this store plugin. But do not rename the anomaly-detection plugin itself. That just uses the renamed GeoLocationStore module/plugin

File: packages/anomaly-detection-store/README.md (hunk 2 above)
Comment (zoemaas) [reply packages/anomaly-detection-store/README.md:15]: Replaced Veramo with Sphereon

File: packages/anomaly-detection-store/README.md (hunk 4 above)
Comment (zoemaas) [reply packages/anomaly-detection-store/README.md:40]: geolocation is a single word: https://dictionary.cambridge.org/dictionary/english/geolocation. If we want to prefix the method with geo, doesn't it make more sense to call it geolocationLookup?

File: packages/anomaly-detection-store/README.md (hunk 3 above)
Comment (zoemaas) [reply packages/anomaly-detection-store/README.md:30]: geolocation is a single word: https://dictionary.cambridge.org/dictionary/english/geolocation. Doesn't it make more sense to rename it to GeolocationStore?

File: packages/anomaly-detection-store/README.md (hunk 4 above)
Comment (nklomp) [reply packages/anomaly-detection-store/README.md:40]: yes

File: packages/anomaly-detection-store/src/agent/AnomalyDetectionStore.ts (hunk 5 above)
Comment (nklomp) [reply packages/anomaly-detection-store/src/agent/AnomalyDetectionStore.ts:28]: of course

**Pull request merged**